packages/amplify-provider-awscloudformation/lib/attach-backend.js

```diff
@@ -0,0 +1,299 @@
+      .promise();
+    apps = apps.concat(listAppsResponse.apps);
+  } while (listAppsResponse.nextToken);
+
+  if (apps.length > 1) {
+    const options = [];
+    apps.forEach(app => {
+      const option = {
+        name: `${app.name} / ${app.appId}`,
+        value: app,
+        short: app.appId,
+      };
+      options.push(option);
+    });
+
+    const { selection } = await inquirer.prompt({
+      type: 'list',
+      name: 'selection',
+      message: 'Which app are you working on?',
+      choices: options,
+    });
+
+    return selection;
+  } else if (apps.length === 1) {
+    context.print.info(`Found one amplify project '${apps[0].name} / ${apps[0].appId}'`);
+    const confirmMigrateMessage = 'Do you want to choose it?';
+    if (await context.prompt.confirm(confirmMigrateMessage)) {
+      return apps[0];
+    }
+  } else {
+    const errorMessage = `Found no currently existing amplify project.`;
+    context.print.error(errorMessage);
+    throw new Error(errorMessage);
+  }
+}
+
+async function getBackendEnv(context, amplifyClient, amplifyApp) {
+  // If envName is in the inputParams, verify it
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify && inputParams.amplify.envName) {
+    const inputEnvName = inputParams.amplify.envName;
+    try {
+      const getBackendEnvironmentResult = await amplifyClient
+        .getBackendEnvironment({
+          appId: amplifyApp.appId,
+          environmentName: inputEnvName,
+        })
+        .promise();
+      context.print.info(`Input envName ${inputEnvName} is verified in Amplify service project ${amplifyApp.name}.`);
+      return getBackendEnvironmentResult.backendEnvironment;
+    } catch (e) {
+      context.print.error(`Input envName ${inputEnvName} is invalid in Amplify service project ${amplifyApp.name}.`);
+      context.print.info(e);
+      throw e;
+    }
+  }
+
+  // If envName is not in the inputParams, prompt user to select
+  let backendEnvs = [];
+  let listEnvResponse = {};
+  do {
+    listEnvResponse = await amplifyClient
+      .listBackendEnvironments({
+        appId: amplifyApp.appId,
+        nextToken: listEnvResponse.nextToken,
+      })
+      .promise();
+
+    backendEnvs = backendEnvs.concat(listEnvResponse.backendEnvironments);
+  } while (listEnvResponse.nextToken);
+
+  if (backendEnvs.length > 1) {
+    const options = [];
+    backendEnvs.forEach(env => {
+      const option = {
+        name: env.environmentName,
+        value: env,
+        short: env.environmentName,
+      };
+      options.push(option);
+    });
+
+    const { selection } = await inquirer.prompt({
+      type: 'list',
+      name: 'selection',
+      message: 'Pick a backend environment:',
+      choices: options,
+    });
+
+    return selection;
+  } else if (backendEnvs.length === 1) {
+    context.print.info(`Found one backend environment '${backendEnvs[0].environmentName}'`);
+    const confirmMigrateMessage = 'Do you want to choose it?';
```

Comment:
Same as above for "choose"

Reply:
wording is decided by the PM. And it's not `merge`, attach-backend workflow is only reached when the user's project does not have a backend project.

Reply:
The recommendation was for when the user only has one Amplify app to automatically select it. The language above is different from what I recommended. "Backend environment '$envname' found. Initializing.." and to get rid of the 'Do you want to choose it'

Reply:
changed